NorthstarLauncher/dllmain.cpp

```diff
@@ -3,48 +3,25 @@
 #include <shlwapi.h>
 #include <filesystem>
 
-HINSTANCE hLThis = 0;
+#include "util.h"
+
 FARPROC p[857];
 HINSTANCE hL = 0;
 
-bool GetExePathWide(wchar_t* dest, DWORD destSize)
-{
-	if (!dest)
-		return NULL;
-	if (destSize < MAX_PATH)
-		return NULL;
-
-	DWORD length = GetModuleFileNameW(NULL, dest, destSize);
-	return length && PathRemoveFileSpecW(dest);
-}
-
-wchar_t exePath[4096];
-wchar_t buffer1[8192];
-wchar_t buffer2[12288];
-
 BOOL WINAPI DllMain(HINSTANCE hInst, DWORD reason, LPVOID)
 {
 	if (reason == DLL_PROCESS_ATTACH)
 	{
-		hLThis = hInst;
-
-		if (!GetExePathWide(exePath, 4096))
-		{
-			MessageBoxA(
-				GetForegroundWindow(),
-				"Failed getting game directory.\nThe game cannot continue and has to exit.",
-				"Northstar Wsock32 Proxy Error",
-				0);
-			return true;
-		}
-
-		SetCurrentDirectoryW(exePath);
+		SetCurrentDirectoryW(GetExePathWide());
 
 		if (!ProvisionNorthstar()) // does not call InitialiseNorthstar yet, will do it on LauncherMain hook
 			return true;
 
+		wchar_t buffer1[4096 * 2];
+		wchar_t buffer2[4096 * 3];
```

Comment:
Hm, Visual Studio/MSVC will warn you if your stack size in a single function exceeds 16384. Technically this function doesn't go deep in chain and our stack size is big enough. At the same time I get it's an improvement over wasting heap space forever I guess, but it's really not ideal either way.

If we don't care about heap space waste, then it'd be the easiest to just change them to static probably.

At the same time, the original code was written to avoid any allocations. That constraint was already violated in several places, and wasn't really probably actually really needed to begin with, so the whole code here could probably be refactored to use std::wstrings and stuff perhaps?

If you include `<fmt/xchar.h>` from spdlog's fmt (if it's not already), you can probably also use `fmt::format(L"{} etc", ...)`

Reply:
Well, you moved the commented lines from heap to stack. If you're okay with the compiler warning as a result of that, fine I guess...?

Reply:
right, completely forgot about that.
I mean, it *shouldn't* cause issues, but it would be better to just move it to be static.

Reply:
thought about it
best to just turn static for now and implement a more proper solution later.

Also went and turned some other globals static, to ensure they don't leak across translation units for some reason.